src/App.php

```diff
@@ -49,15 +50,23 @@
 class App extends AbstractApp {
 
 	/**
-	 * @var array Languages supported by the app
-	 */
-	public $supportedLanguages = [ 'en', 'fr' ];
-
-	/**
-	 * @return array Languages supported by the app
+	 * Get codes of all langages supported by CopyPatrol. This list is retrieved from the
+	 * copyright_diffs database table, and cached for a week.
+	 * @return string[] The 2-letter language codes.
 	 */
 	public function getSupportedLanguages() {
-		return $this->supportedLanguages;
+		// Use the cached list if possible.
+		$cacheItem = $this->slim->cache->getItem( 'supported-languages' );
+		if ( $cacheItem->isHit() ) {
+			return $cacheItem->get();
+		}
+
+		// Otherwise get the list from the database, and cache it for a week.
+		$langs = $this->getPlagiabotDao()->getLanguages();
+		$cacheItem->expiresAfter( new DateInterval( 'P7D' ) );
```

Comment:
maybe also explain what `P7D` is. I'm safely guessing 7D = 7 days, but what does the P do?